src/schroed_dirac_solver.f90

```diff
@@ -23,16 +23,17 @@ subroutine total_energy(Z, rmax, Ne, a, p, Nq, DOFs, alpha_int, dirac_int, &
     integer, intent(out) :: DOFs
     real(dp), allocatable, intent(out) :: lam(:), eigfn(:,:,:)
     real(dp), intent(out) :: xq(Nq, Ne)
-    real(dp), allocatable :: xe(:), xiq(:), wtq(:), V(:,:), xin(:), &
+    real(dp), allocatable :: xe(:), xiq(:), wtq(:), V(:,:), Vin(:,:), xin(:), &
         H(:,:), S(:), S2(:,:), DSQ(:), phihq(:,:), dphihq(:,:), xiq_lob(:), wtq_lob(:), &
-        D(:,:), lam2(:), xiq1(:), wtq1(:), xq1(:,:), fullc(:), uq(:,:), rho(:,:)
-    integer, allocatable :: in(:,:), ib(:,:)
+        D(:,:), lam2(:), xiq1(:), wtq1(:), xq1(:,:), fullc(:), uq(:,:), rho(:,:), &
+        focc(:,:), lam_tmp(:), rho1(:,:), xiq_gj(:,:), wtq_gj(:,:)
+    integer, allocatable :: in(:,:), ib(:,:), focc_idx(:,:)
     real(dp) :: rmin
-    integer :: al, i, j, l, k, ind, kappa, Nb, Nn
+    integer :: al, i, j, l, k, ind, kappa, Nb, Nn, idx, Lmin2, Lmax
     real(dp) :: E_dirac_shift
     rmin = 0
     allocate(xe(Ne+1), xiq(Nq), xiq1(Nq), wtq(Nq), wtq1(Nq), V(Nq, Ne), xiq_lob(p+1), wtq_lob(p+1))
-    allocate(phihq(Nq, p+1), dphihq(Nq, p+1), in(p+1, Ne), ib(p+1, Ne), xin(p+1), xq1(Nq, Ne))
+    allocate(phihq(Nq, p+1), dphihq(Nq, p+1), in(p+1, Ne), ib(p+1, Ne), xin(p+1), xq1(Nq,1))
```

Comment:
1 is correct here, consistent with DFT. Inside `solve_dirac_eigenproblem` only the first element is used. This should eventually be converted to just a 1D array.